Build Error from #257

We just got a dependabot PR into XDP (https://github.com/microsoft/xdp-for-windows/pull/92) to update our submodule for wil and it's [failing to build](https://github.com/microsoft/xdp-for-windows/runs/8281952556?check_suite_focus=true) because of #257:
```
Build FAILED.

       "D:\a\xdp-for-windows\xdp-for-windows\xdp.sln" (default target) (1) ->
       "D:\a\xdp-for-windows\xdp-for-windows\test\functional\lib\xdptests.vcxproj" (default target) (19) ->
       (RunNativeCodeAnalysis target) -> 
         D:\a\xdp-for-windows\xdp-for-windows\submodules\wil\include\wil\result_macros.h(1314): error C6387: 'Temp_value_#1642' could be '0':  this does not adhere to the specification for the function 'GetProcAddress'.  [D:\a\xdp-for-windows\xdp-for-windows\test\functional\lib\xdptests.vcxproj]
```
I'm not 100% sure why, but it doesn't like this line:
```c
pfnRtlDisownModuleHeapAllocation = reinterpret_cast<decltype(pfnRtlDisownModuleHeapAllocation)>(::GetProcAddress(::GetModuleHandleW(L"ntdll.dll"), "RtlDisownModuleHeapAllocation"));
```

--- a/include/wil/result_macros.h
+++ b/include/wil/result_macros.h
@@ -1313,7 +1313,7 @@ namespace wil
             {
                 if (auto ntdllModule = ::GetModuleHandleW(L"ntdll.dll"))
                 {
-                    pfnRtlDisownModuleHeapAllocation = reinterpret_cast<decltype(pfnRtlDisownModuleHeapAllocation)>(::GetProcAddress(ntdllModule, "RtlDisownModuleHeapAllocation");
+                    pfnRtlDisownModuleHeapAllocation = reinterpret_cast<decltype(pfnRtlDisownModuleHeapAllocation)>(::GetProcAddress(ntdllModule, "RtlDisownModuleHeapAllocation"));
                 }
                 fetchedRtlDisownModuleHeapAllocation = true;
 

--- a/include/wil/result_macros.h
+++ b/include/wil/result_macros.h
@@ -1311,7 +1311,10 @@ namespace wil
             }
             else if (!fetchedRtlDisownModuleHeapAllocation)
             {
-                pfnRtlDisownModuleHeapAllocation = reinterpret_cast<decltype(pfnRtlDisownModuleHeapAllocation)>(::GetProcAddress(::GetModuleHandleW(L"ntdll.dll"), "RtlDisownModuleHeapAllocation"));
+                if (auto ntdllModule = ::GetModuleHandleW(L"ntdll.dll"))
+                {
+                    pfnRtlDisownModuleHeapAllocation = reinterpret_cast<decltype(pfnRtlDisownModuleHeapAllocation)>(::GetProcAddress(ntdllModule, "RtlDisownModuleHeapAllocation");
+                }
                 fetchedRtlDisownModuleHeapAllocation = true;
 
                 if (pfnRtlDisownModuleHeapAllocation)

